siakg/main.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+// main.go defines the command structure using the cobra package. This includes
+// configuration variables and flags.
+
+import (
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	// Version of the siakg program.
+	Version       = "1.0"
+	FileExtension = ".siakey"
+
+	DefaultFolder       = "SiafundKeys"
+	DefaultKeyname      = "SiafundKeys"
+	DefaultRequiredKeys = 2
+	DefaultTotalKeys    = 3
+)
+
+var (
+	// A global variable containing all of the configuration information,
+	// necessary for interacting with cobra.
+	config Config
+)
+
+// The Config struct holds all of the configuration variables. The format is
+// made to be compatible with gcfg. gcfg is not currently used in the siakg
+// project, however it helps maintain consistency with the design of siad.
+type Config struct {
+	Siakg struct {
+		Folder       string
+		KeyName      string
+		RequiredKeys int
+		TotalKeys    int
+	}
+
+	Address struct {
+		Filename string
+	}
+}
```

Comment:
I think in the case of `KeyInfo` it makes more sense to make the filename a required argument, rather than a flag. Probably true of a few other flags as well, like `RequiredKeys` and `TotalKeys`.